doc/reference/random/index.rst

```diff
@@ -0,0 +1,96 @@
+.. _random_reference:
+
+Random
+######
+
+.. toctree::
```

Comment:
The ``toctree`` directive is to indicate files that should be added to the table of contents for the site.  I suspect you're trying to add a local table of contents listing for the headings in this specific document, so you should instead use:
```
.. contents::
   :depth: 2
```